internal/cqarrow/mac.go

```diff
@@ -0,0 +1,189 @@
+package cqarrow
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/apache/arrow/go/v12/arrow/memory"
+	"github.com/goccy/go-json"
+)
+
+type MacBuilder struct {
+	*array.ExtensionBuilder
+	dtype *MacType
+}
+
+func NewMacBuilder(mem memory.Allocator, dtype arrow.ExtensionType) *MacBuilder {
+	b := &MacBuilder{
+		ExtensionBuilder: array.NewExtensionBuilder(mem, dtype),
+		dtype:            dtype.(*MacType),
+	}
+	return b
+}
+
+func (b *MacBuilder) Append(v net.HardwareAddr) {
+	b.ExtensionBuilder.Builder.(*array.BinaryBuilder).Append(v[:])
+}
+
+func (b *MacBuilder) UnsafeAppend(v net.HardwareAddr) {
+	b.ExtensionBuilder.Builder.(*array.BinaryBuilder).UnsafeAppend(v[:])
+}
+
+func (b *MacBuilder) AppendValues(v []net.HardwareAddr, valid []bool) {
+	data := make([][]byte, len(v))
+	for i, v := range v {
+		data[i] = v
+	}
+	b.ExtensionBuilder.Builder.(*array.BinaryBuilder).AppendValues(data, valid)
+}
+
+func (b *MacBuilder) UnmarshalOne(dec *json.Decoder) error {
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	var val net.HardwareAddr
+	switch v := t.(type) {
+	case string:
+		data, err := net.ParseMAC(v)
+		if err != nil {
+			return err
+		}
+		val = data
+	case []byte:
+		val = net.HardwareAddr(v)
+	case nil:
+		b.AppendNull()
+		return nil
+	default:
+		return &json.UnmarshalTypeError{
+			Value:  fmt.Sprint(t),
+			Type:   reflect.TypeOf([]byte{}),
+			Offset: dec.InputOffset(),
+			Struct: "Binary",
+		}
+	}
+
+	b.Append(val)
+	return nil
+}
+
+func (b *MacBuilder) Unmarshal(dec *json.Decoder) error {
+	for dec.More() {
+		if err := b.UnmarshalOne(dec); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *MacBuilder) UnmarshalJSON(data []byte) error {
+	dec := json.NewDecoder(bytes.NewReader(data))
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	if delim, ok := t.(json.Delim); !ok || delim != '[' {
+		return fmt.Errorf("fixed size binary builder must unpack from json array, found %s", delim)
+	}
+
+	return b.Unmarshal(dec)
+}
+
+// UUIDArray is a simple array which is a FixedSizeBinary(16)
```

Comment:
Same here: It looks like all the comments in this file that refer to FixedSizeBinary or UUID need to be updated (they were copied from UUID)